packages/suite/src/actions/suite/metadataActions.ts

```diff
@@ -730,6 +1022,36 @@ export const init =
 
             const providerResult = await dispatch(initProvider());
             if (!providerResult) {
+                dispatch(abortInit());
+                return false;
+            }
+        }
+
+        // 4. get the metadata minimum version to determine a migration is needed
```

Comment:
just throwing some ideas around. this migration-needed-decision logic maybe could be encapsulated somewhere and within `init` action only 2 methods could be called something like `isMigrationNeeded` `migrate`. Unsure yet how exactly I would like it 😄 will keep thinking about it.

Reply:
done